oscore-cf/src/main/java/org/eclipse/californium/oscore/CoapOSExceptionHandler.java

```diff
@@ -102,11 +105,19 @@ public static EmptyMessage manageError(OSException e, Response response) {
 			LOGGER.error(ErrorDescriptions.ERROR_MESS_NULL);
 			throw new NullPointerException(ErrorDescriptions.ERROR_MESS_NULL);
 		}
+		
+		//Reply with empty ACK when decryption of Response fails
+		if(errMess.equals(ErrorDescriptions.DECRYPTION_FAILED)) {
+			LOGGER.info("Sending empty ACK message due to Response decryption failure");
+			return EmptyMessage.newACK(response);
```

Comment:
RFC7252 responses may be piggybacked. If so, a ACK for a ACK is not compliant.
Dose OSCORE change this? Are piggybacked responses not used with OSCORE?

Reply:
I guess, it should be moved behind the 

```
 		if (!response.isConfirmable()) {
 			LOGGER.error("An Empty Message will not be created");
 			return null;
 		}
```

Reply:
This functionality stemmed from a testing specification I was running some tests according to. I will look into this point further.

Reply:
For now I have removed these lines until I can look into this further. In such case I can add them in a future pull request again.

Removed in commit https://github.com/rikard-sics/californium/commit/7e96b608bb45bb7204c788323f222c268921af96